components/scream/src/physics/p3/atmosphere_microphysics.cpp

```diff
@@ -300,12 +312,14 @@ void P3Microphysics::initialize_impl (const RunType /* run_type */)
   history_only.vap_liq_exchange = get_field_out("micro_vap_liq_exchange").get_view<Pack**>();
   history_only.vap_ice_exchange = get_field_out("micro_vap_ice_exchange").get_view<Pack**>();
   // -- Set values for the post-amble structure
-  p3_postproc.set_variables(m_num_cols,nk_pack,prog_state.th,pmid,T_atm,t_prev,
-      prog_state.qv, prog_state.qc, prog_state.nc, prog_state.qr,prog_state.nr,
-      prog_state.qi, prog_state.qm, prog_state.ni,prog_state.bm,qv_prev,
-      diag_outputs.diag_eff_radius_qc,diag_outputs.diag_eff_radius_qi,
-      diag_outputs.precip_liq_surf,diag_outputs.precip_ice_surf,
-      precip_liq_surf_mass,precip_ice_surf_mass);
+  p3_postproc.set_variables(m_num_cols,nk_pack,
+                            prog_state.th,pmid,T_atm,t_prev,
+                            prog_state.qv, prog_state.qc, prog_state.nc, prog_state.qr,prog_state.nr,
+                            prog_state.qi, prog_state.qm, prog_state.ni,prog_state.bm,qv_prev,
+                            diag_outputs.diag_eff_radius_qc,diag_outputs.diag_eff_radius_qi,
+                            diag_outputs.precip_liq_surf,diag_outputs.precip_ice_surf,
+                            precip_liq_surf_mass,precip_ice_surf_mass,
+                            vapor_flux, water_flux, ice_flux, heat_flux);
```

Comment:
If these fields are not always present, you should find a different way to pass them. Perhaps using pointers, which are null if fields are not there?